### PR TITLE
Consolidate Dockerfiles

### PR DESCRIPTION
**Pull request by 0snap**
###  :notebook_with_decorative_cover: Description

- [x] consolidate Dockerfiles
- [x] update CI workflow to test Dockerfile
- [ ] update general CI workflow

###  :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

**Review by dominiklohmann — COMMENTED** [review]
We must factor in the changes from https://github.com/tenzir/vast/pull/1310 as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- ⚡️ The GitHub CI now runs on Debian Buster and produces Debian artifacts.
+  Similarly, the Docker images we provide on
+  [dockerhub](https://hub.docker.com/r/tenzir/vast) use Debian Buster as
+  base-image.
+  [#1294](https://github.com/tenzir/vast/pull/1294)
```
Comment (mavam) [inline CHANGELOG.md:19]: Could we add sentence fragment about what we had before? Not everyone might be aware of that.

**Comment by 0snap** [comment]
@dominiklohmann CI turned green and I verified the GCC-8 artifact as follows:
- download the zip (`vast-2020.12.16-363-g16ee0a423-linux-release-gcc.tar.gz.zip`)
- unpack it inside the vast repo to a (sub-)folder called `opt/vast`
- run `DOCKER_BUILDKIT=1 docker build . -t vast-build-pre --build-arg DOCKER_BUILD=prebuilt`
- test the prebuilt image: `docker run vast-build-pre` -> prints vast help

**Comment by dominiklohmann** [comment]
Thanks for the instructions. I'll try this locally shortly, and will also try to confirm the completeness of the VAST installation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,13 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- ⚡️ The GitHub CI now runs on Debian Buster and produces Debian artifacts
+  instead of Ubuntu artifacts. Similarly, the Docker images we provide on
+  [dockerhub](https://hub.docker.com/r/tenzir/vast) use Debian Buster as
+  base-image. To build Docker images locally, users now must set
```
Comment (mavam) [inline CHANGELOG.md:18]:
```suggestion
  base image. To build Docker images locally, users now must set
```

Just a low-prio nit. If this is technical Docker term, you also do `base-image`.

File: CHANGELOG.md (hunk 1 above)
Comment (mavam) [reply CHANGELOG.md:20]: Thanks, reads much more understandable now!

File: CHANGELOG.md (hunk 2 above)
Comment (0snap) [reply CHANGELOG.md:18]: The official Docker docs spell it without dash: https://docs.docker.com/develop/develop-images/baseimages/

Comment (dominiklohmann) [reply CHANGELOG.md:18]: Fwiw if you don't wanna do this rn, I'll do a pass over all items for the release prep anyways.

**Review by dominiklohmann — APPROVED** [review]
I tested this extensively. Everything works as expected, and I think this is a major improvement that is worth the additional effort to maintain our CI.

**Pull request merged**